exporter/lokiexporter/factory_next.go

```diff
@@ -0,0 +1,39 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lokiexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/lokiexporter"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+)
+
+func createNextLogsExporter(ctx context.Context, set component.ExporterCreateSettings, cfg *Config) (component.LogsExporter, error) {
+	exp := newNextExporter(cfg, set.TelemetrySettings)
+
+	return exporterhelper.NewLogsExporter(
```

Comment:
You mean, because of the removal of the attributes? Other than that, we shouldn't be mutating the data. In fact, I would even argue that we shouldn't change the original data at all. I'll change it to reflect the current state, but we likely want to avoid changing the input data.

Reply:
I ended up making copies of the resource and log attributes, so that this component is not mutating the incoming data anymore.

Reply:
FYI: That is less efficient. marking mutating will affect performance only if you have multiple pipelines, otherwise the collector is smart enough to not do extra copies.

Reply:
Got it, I'll leave it like that and revert if we see performance degradation high enough to warrant mutating the data in the component. I believe that having it semantically correct is preferable to having it with the best performance possible if they contradict each other.